### PR TITLE
Feat react flow renderer

### PR DESCRIPTION
**Pull request by BastinJafari**
Implement a new library for flow charts

### CODE REVIEW

File: src/components/FlowChart/ChartNode.tsx [hunk 1]
```diff
@@ -1,50 +1,87 @@
...
-      </div>
-    </div>
-  );
+const NodePortTop = styled(Handle)`
+  margin-top: -0.2rem;
+  background-color: white;
+  width: 1rem;
+  height: 1rem;
+  border-radius: 0.5rem;
+  border: 1px solid rgba(0, 153, 153, 0.3); //todo use theming
```
Comment (Roshanjossey) [inline src/components/FlowChart/ChartNode.tsx:28]: Please sue colors from theme. There's a bunch of colors around this.

File: src/components/FlowChart/ChartNode.tsx [hunk 2]
```diff
@@ -1,50 +1,87 @@
...
+        const type = useSelector(selectSelectedFlow).type
+        const dispatch = useDispatch()
+        return (
+          <ChartNodeElement
+            onDoubleClick={() => {
+              type === "user-generated" &&
+                dispatch(showModal("podEdit", { nodeId: id }))
+            }}
+          >
+            <NodePortTop type="target" position={Position.Top} />
```
Comment (Roshanjossey) [inline src/components/FlowChart/ChartNode.tsx:63]: Most of what needs to be rendered for `Pod` and `Gateway` looks similar except for `<NodePortTop>`. If this is the only difference, how about making only this part conditional?

File: src/helpers/flow-chart.ts [hunk 3]
```diff
@@ -1,142 +1,131 @@
 // @ts-nocheck
```
Comment (Roshanjossey) [inline src/helpers/flow-chart.ts:1]: Are we god to remove `ts-nocheck` from this file now?

**Review by Roshanjossey — CHANGES_REQUESTED** [review]

File: src/helpers/flow-chart.ts (hunk 3 above)
Comment (BastinJafari) [reply src/helpers/flow-chart.ts:1]: I'll remove it in the other PR about removing todos\\

**Review by Roshanjossey — APPROVED** [review]

**Pull request merged**